packages/language-support/src/parserWrapper.ts

```diff
@@ -68,52 +91,78 @@ export type LabelOrRelType = {
   };
 };
 
-export interface EnrichedParsingResult extends ParsingResult {
-  diagnostics: SyntaxDiagnostic[];
-  stopNode: ParserRuleContext;
-  collectedLabelOrRelTypes: LabelOrRelType[];
-  collectedVariables: string[];
-  collectedCommands: ParsedCommand[];
-}
-
-export interface ParsingScaffolding {
-  query: string;
-  parser: CypherParser;
-  tokenStream: CommonTokenStream;
-}
-
 export function createParsingScaffolding(query: string): ParsingScaffolding {
   const inputStream = CharStreams.fromString(query);
   const lexer = new CypherLexer(inputStream);
   const tokenStream = new CommonTokenStream(lexer);
-  const parser = new CypherParser(tokenStream);
-  parser.removeErrorListeners();
+  const stsTokenStreams = splitIntoStatements(tokenStream, lexer);
+
+  const statementsScaffolding: StatementParsingScaffolding[] =
+    stsTokenStreams.map((t) => {
+      const tokens = [...t.tokens];
```

Comment:
Why doe we need to copy this list? Is there some mutation going on later or is it just to be on the safe side?

Reply:
It surprised me but when splitting the token stream into separate ones, the parser can sometimes modify some of them and add an extra EOF, which I would not expect, so it's just a protection against the parser modifying the tokens